NavigationAdvancedSample/app/src/main/java/com/example/android/navigationadvancedsample/conductor/ControllerNavigator.kt

```diff
@@ -36,99 +33,13 @@ import kotlin.collections.set
 class ControllerNavigator(private val router: Router) :
     Navigator<ControllerNavigator.Destination>() {
 
-    private val handler = Handler()
```

Comment:
@chc-unwire TBH, I can't see how this would fly.. I specifically spent a lot of time figuring out this part. Perhaps check the old commit msgs.
The concept of a "desired backstack" is to allow the navigation abstraction to be completely be in control of the backstack - not relying on Conductor's (router) managing the back stack. This is key, since conductor doesn't know about non-conductor destinations. The Routers should simple be told to set their back stack - to keep in sync with the navigation abstraction backstack.

Perhaps I am missing something else you did here?

Reply:
Let's talk again tomorrow, but I re-tested the deeplinking (albeit not fully tested as I dont have a proper deeplinking flow implemented in this demo app, should test that properly though), the current implementation which relies on the shared ConductorNavHost with different graphs inflated, requires each graph to be inflated in turn, in order to ask if it can handle any given deeplink. Doing this means inflating and replacing several graphs in quick succession, and as such any async behaviour in managing this backstack, leads to issues, where the graph and backstacks arent in sync, and i see crashes such as:
`2021-01-25 20:21:08.491 21356-21356/com.example.android.navigationadvancedsample E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.android.navigationadvancedsample, PID: 21356
    java.lang.IllegalArgumentException: navigation destination com.example.android.navigationadvancedsample:id/action_title_to_about is unknown to this NavController
        at androidx.navigation.NavController.navigate(NavController.java:892)
        at androidx.navigation.NavController.navigate(NavController.java:833)
        at androidx.navigation.NavController.navigate(NavController.java:819)
        at androidx.navigation.NavController.navigate(NavController.java:807)
        at com.example.android.navigationadvancedsample.conductor.homescreen.Title$onCreateView$1.onClick(Title.kt:30)
        at android.view.View.performClick(View.java:7140)
        at com.google.android.material.button.MaterialButton.performClick(MaterialButton.java:967)
        at android.view.View.performClickInternal(View.java:7117)
        at android.view.View.access$3500(View.java:801)
        at android.view.View$PerformClick.run(View.java:27351)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7356)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)`
        
The two other "tougher" scenarios, restoring app backstacks from "don't keep activities enabled", and view clicks being responsive after device orientation changes, are "fixed" by doing two reflected router destroy calls, which I'll push tomorrow as well for you to play around with and break :)